### PR TITLE
add 'relative positional embedding (RPE)' feature - re-creating after…

### PR DESCRIPTION
**Pull request by khcs**
… re-basing to 'main' branch from r1.9.0 - all previous review comments are in.

Signed-off-by: Hoo Chang Shin <hshin@nvidia.com>

# What does this PR do ?

Add 'relative positional embedding (RPE)' feature to T5 model.

**Collection**: NLP

# Changelog 
Adds an option to use 'relative positional embedding' other than 'absolute position embedding' or 'fixed position encoding'.

# Usage

In the 'megatron_t5_config.yaml' file change the 'position_embedding_type' option to 'relative'.

```python
python megatron_t5_pretraining.py
```

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/nlp/modules/common/megatron/token_level_encoder_decoder.py [hunk 1]
```diff
@@ -123,6 +126,17 @@ def __init__(
         self.add_encoder = add_encoder
         self.add_decoder = add_decoder
         self.normalization = normalization
+        self.position_embedding_type = position_embedding_type
+        self.relative_attention_num_buckets = relative_attention_num_buckets
+        self.relative_attention_max_distance = relative_attention_max_distance
+
+        if self.position_embedding_type == 'absolute' or 'learned_absolute':
```
Comment (MaximumEntropy) [inline nemo/collections/nlp/modules/common/megatron/token_level_encoder_decoder.py:133]: We don't support just `absolute` yet, so maybe remove it from this check?

Comment (MaximumEntropy) [reply nemo/collections/nlp/modules/common/megatron/token_level_encoder_decoder.py:133]: And in the error raise below

**Review by MaximumEntropy — APPROVED** [review]
LGTM. Just minor comments.

File: nemo/collections/nlp/modules/common/megatron/token_level_encoder_decoder.py (hunk 1 above)
Comment (khcs) [reply nemo/collections/nlp/modules/common/megatron/token_level_encoder_decoder.py]: yes, removed them.

**Review by MaximumEntropy — DISMISSED** [review]
LGTM!

**Review by MaximumEntropy — APPROVED** [review]

**Pull request merged**